homeassistant/components/livisi/coordinator.py

```diff
@@ -0,0 +1,139 @@
+        for device in shc_devices:
+            if device in self.devices:
+                continue
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in self.switch_devices:
+                    self.switch_devices.append(device)
+                    async_dispatcher_send(self.hass, LIVISI_DISCOVERY_NEW, device)
+            self.devices.append(device)
+
+    async def async_setup(self) -> None:
+        """Set up the Livisi Smart Home Controller."""
+        if bool(self.aiolivisi.livisi_connection_data) is False:
+            livisi_connection_data = {
+                "ip_address": self.config_entry.data["host"],
+                "password": self.config_entry.data["password"],
+            }
+
+            await self.aiolivisi.async_set_token(
+                livisi_connection_data=livisi_connection_data
+            )
+        controller_info = await self.aiolivisi.async_get_controller()
+
+        controller_data = controller_info
+        if controller_data.get("controllerType") == "Avatar":
+            self.port = AVATAR_PORT
+            self.is_avatar = True
+        else:
+            self.port = CLASSIC_PORT
+            self.is_avatar = False
+        self.serial_number = controller_data.get("serialNumber")
+        self.controller_type = controller_data.get("controllerType")
+        self.os_version = controller_data.get("osVersion")
+
+    async def async_get_devices(self) -> list:
+        """Set the discovered devices list."""
+        shc_devices = await self.aiolivisi.async_get_devices()
+        if bool(shc_devices) is False:
+            shc_devices = await self.aiolivisi.async_get_devices()
+        return shc_devices
+
+    async def async_get_pss_state(self, capability):
+        """Set the PSS state."""
+        response = await self.aiolivisi.async_get_pss_state(capability[1:])
+        if response is None:
+            return
+        on_state = response.get("onState")
+        return on_state.get("value")
+
+    async def async_set_all_rooms(self):
+        """Set the room list."""
+        response = await self.aiolivisi.async_get_all_rooms()
+
+        for available_room in response:
+            self.rooms[available_room.get("id")] = available_room.get("config").get(
+                "name"
+            )
+
+    def on_data(self, event_data: LivisiEvent) -> None:
+        """Define a handler to fire when the data is received."""
+        source = event_data.source
+        if event_data.onState is not None:
+            device_id_state: dict = {
+                "id": source,
+                "state": event_data.onState,
+            }
+            self.async_set_updated_data(device_id_state)
+        if event_data.isReachable is not None:
+            device_id_reachability: dict = {
+                "id": source.replace("/device/", ""),
+                "is_reachable": event_data.isReachable,
+            }
+            self.async_set_updated_data(device_id_reachability)
```

Comment:
We can't set different kind of data as coordinator data. The coordinator data should always represent the same thing. It should normally be the state of the devices or service. If there's some different state we want to track we can't do that in the same coordinator data.

Reply:
Yes, we can use the coordinator to track the one kind of state and our dispatch helper to signal for other state updates. It's also possible to create multiple different coordinators to track different states that represent different things.

Reply:
I modified this according to your comment, thank you. The device configuration is using the polling coordinator to update. The states of the device are updated by using the `dispatcher_send` method and the reachability of the device is updated by `dispatcher_send` ,